File: result/repair/starchat/java2python/java_repair/FORM_MINIMUM_NUMBER_FROM_GIVEN_SEQUENCE_1.java
  String f_gold ( String seq ) {
        int n = seq. length ();
        if ( n >= 9 ) return "-1";
        String [] result = new String [ n + 1 ];
        int count = 1;
        for ( int i = 0; i <= n; i++ ) {
            if ( i == n || seq. charAt ( i ) == 'I' ) {
                for (int j=i-1;j>-2;j--){
                    result [ j + 1 ] = ( char ) ( '0' + count );
                    count++;
                    if ( j >= 0 && seq. charAt ( j ) == 'I' ) break;
                }
            }
        }
        return String. join ( "", result );
    }

File: result/translate/starchat/java2cpp/c++/FORM_MINIMUM_NUMBER_FROM_GIVEN_SEQUENCE_1.cpp
string f_gold ( string seq ) {
        int n = seq.length();
        if (n >= 9) return "-1";
        char result[n + 1];
        int count = 1;
        for (int i = 0; i <= n; i++) {
            if (i == n || seq[i] == 'I') {
                for (int j = i - 1; j >= 0; j--) {
                    result[j + 1] = (char)('0' + count++);
                    if (j >= 0 && seq[j] == 'I') break;
                }
            }
        }
        return string(result);
    }